### PR TITLE
CS #282 - SAML configuration

### PR DESCRIPTION
**Pull request by dleadbetter**
This pull request adds the SAML configuration (via omniauth-saml) and adds the domain configuration for purdue and purdue-staging.

**SAML Login (via One Login)**
cove_studio_saml

**Email Login**
cove_studio_email

**ActiveAdmin Tenant Configuration**
Screen Shot 2020-09-03 at 2 20 13 PM

### CODE REVIEW

File: Procfile.dev [hunk 1]
```diff
@@ -1,3 +1,3 @@
-web: bundle exec unicorn -p $PORT -c ./config/unicorn.rb
```
Comment (dleadbetter) [inline Procfile.dev:1]: @jamiefolsom - I changed this to use `puma`, as `unicorn` didn't appear to be included in the `Gemfile`.

**Comment by jamiefolsom** [comment]
@dleadbetter this needs to be rebased on master -- another PR was merged to master. Thanks!

**Review by jamiefolsom — APPROVED** [review]
@dleadbetter this looks great -- let's go ahead and merge, and we can make changes if/as necessary on future branches.

File: app/controllers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -10,15 +13,49 @@ def wordpress_hosted
...
+    if @tenant.present?
+      Rails.logger.info("**********Setup Started for #{@tenant.database_name}************")
+      Rails.logger.info("The tenant auth key is #{@tenant.wp_auth_key}")
+      Rails.logger.info("The tenant secret is #{@tenant.wp_auth_secret}")
+      Rails.logger.info("The tenant url is #{@tenant.wp_url}")
+    end
+
+    request.env['omniauth.strategy'].options[:idp_cert_fingerprint] = idp_cert_fingerprint(@tenant)
+    request.env['omniauth.strategy'].options[:idp_sso_target_url] = idp_sso_target_url(@tenant)
+    request.env['omniauth.strategy'].options[:issuer] = url_for(action: 'metadata', controller: 'saml')
```
Comment (dleadbetter) [inline app/controllers/omniauth_callbacks_controller.rb:57]: @jamiefolsom - Added this to include the `issuer` property as `<current-tentant-url>/sp/metadata`.

Example: `https://purdue-staging.covecollective.org/sp/metadata`.

Comment (jamiefolsom) [reply app/controllers/omniauth_callbacks_controller.rb:57]: Makes sense, although I am now getting a 404 response for [that request](https://purdue-staging.covecollective.org/sp/metadata): 
Screen Shot 2020-09-23 at 09 55 52

Comment (dleadbetter) [reply app/controllers/omniauth_callbacks_controller.rb:57]: Hmm, interesting. I haven't pushed anything to staging yet. I'll take a look.

File: app/views/saml/purdue_metadata.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            k7z84lmRJxrg/NzHsXGXkJSvFeWqrfheonB1lYLDUhSu1xKlsrB2ZeWafHPVCp+L
+            94jUv+wOO0wX3gE8N4xQ/SXSG9U6
+          </ds:X509Certificate>
+        </ds:X509Data>
+      </ds:KeyInfo>
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes256-cbc"/>
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes128-cbc"/>
+      <EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#tripledes-cbc"/>
+    </KeyDescriptor>
+    <AssertionConsumerService index="1" isDefault="true" Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://purdue-staging.covecollective.org/users/auth/saml/callback"/>
```
Comment (dleadbetter) [inline app/views/saml/purdue_metadata.xml:61]: @jamiefolsom - I've added the real callback path here based on the last error information we received.

**Pull request merged**